tests/integration/test_membership.py

```diff
@@ -135,6 +140,29 @@ def test_full_cluster_remove(cluster):
         assert cluster.node(node_id).raft_info()['state'] == 'uninitialized'
 
 
+def test_remove_and_rejoin_node_with_same_id_fails(cluster):
```

Comment:
Full coverage here should consider the different persistence states, i.e. make sure we remember used Ids also if loaded from snapshot, restarted from log, etc.

Reply:
👍